docs/architecture/authentication-and-authorization.md

```diff
@@ -0,0 +1,216 @@
+# Authentication and Authorization
+
+## Introduction
+Currently, communication between the Compass and both Runtimes and Applications are not secured. We need to provide some security possibilities.
+We want to secure the Compass using ORY's Hydra and OathKeeper. There would be three ways of authentication:
+ - OAuth 2.0 
+ - Client Certificates (mTLS)
+ - JWT token issued by Dex
+
+ There would be three possibilities for securing the connection. To achieve that, first, we need to integrate Hydra and OathKeeper into the Compass. We also need to implement additional supporting components to make our scenarios valid.
+
+## Architecture
+
+The following diagram represents the architecture of the security in Compass:
+
+![](./assets/security-architecture.svg)
+
+### Tenant Mapping Handler
+
+It is an OathKeeper [hydrator](https://github.com/ory/docs/blob/525608c65694539384b785355d293bc0ad00da27/docs/oathkeeper/pipeline/mutator.md#hydrator) handler responsible for mapping authentication session data to tenant. It could be built into Director itself, as it will also use the same database. It would be implemented as a separate endpoint, such as `/tenant-mapping`.
+
+To unify the approach for mapping, we could introduce `authorization_id`, widely used by multiple authentication flows.
+The `authorization_id` is equal to:
+- `client_id` in OAuth 2.0 authentication flow 
+- `username` in Basic authentication flow
+- Common Name (CN) in Certificates authentication flow
+
+While generating one-time token, `client_id`/`client_secret` pair or basic authentication details for Runtime/Application/IntegrationSystem (using proper GraphQL mutation on Director), an entry in `system_auths` table in Director database is created. The `system_auths` table is used for tenant mapping.
```

Comment:
I think it would be better to prepare another document regarding IntegrationSystem itself. I don't want to overload this document with too much pieces of information.

For now I will describe Integration System in Terminology doc. Is that OK for you?

Reply:
- Yes:, add to terminology (with proper description how if differs from App and Runtime, e.g. works with multitenant/multiapps/multiruntimes management and exists in global scope without tenant assignment)
- And no: please add to diagram as well. And link to terminology first occurrence.

Reply:
Done